src/component/common/InstanceStatus/InstanceStatus.tsx

```diff
@@ -1,10 +1,35 @@
 import { useInstanceStatus } from 'hooks/api/getters/useInstanceStatus/useInstanceStatus';
-import React, { FC } from 'react';
+import React, { FC, useEffect, useMemo, useState } from 'react';
 import { InstanceStatusBar } from 'component/common/InstanceStatus/InstanceStatusBar';
 import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { Dialogue } from 'component/common/Dialogue/Dialogue';
+import { Typography } from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import { IInstanceStatus, InstanceState } from 'interfaces/instance';
+import { differenceInDays, parseISO } from 'date-fns';
 
 export const InstanceStatus: FC = ({ children }) => {
     const { instanceStatus } = useInstanceStatus();
+    const trialDaysRemaining = calculateTrialDaysRemaining(instanceStatus);
+    const navigate = useNavigate();
+
+    const statusExpired = useMemo(
+        () =>
+            instanceStatus?.state === InstanceState.TRIAL &&
```

Comment:
Might need to check `instanceStatus.plan === pro`

Reply:
Will be covered at a higher level with `isBilling`